### PR TITLE
fix: clarify how to access individual attrs from fetchUserAttributes

### PR DESCRIPTION
**Pull request by richardmcclellan**
A customer on Discord was confused about how to access individual user attributes given that the result returned from `Auth.fetchUserAttributes` is an array.  This PR adds sample code for iOS and Android to convert the array to a map, which allows easier access to the individual attributes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by undefobj — APPROVED** [review]

File: docs/lib/auth/fragments/ios/user_attributes/10_fetch_attributes.md [hunk 1]
```diff
@@ -7,7 +7,14 @@ func fetchAttributes() {
...
         switch result {
         case .success(let attributes):
-            print("User attributes - \(attributes)")
+            print("All user attributes - \(attributes)")
+            var attributesMap = [AuthUserAttributeKey: String]()
+            for attribute in attributes {
+                attributesMap[attribute.key] = attribute.value
+            }
+            if let email = attributesMap[.email] {
+                print("Email is \(email)")
```
Comment (lawmicha) [inline docs/lib/auth/fragments/ios/user_attributes/10_fetch_attributes.md:16]:
```suggestion
                print("User email = \(email)")
```

File: docs/lib/auth/fragments/ios/user_attributes/10_fetch_attributes.md [hunk 2]
```diff
@@ -7,7 +7,14 @@ func fetchAttributes() {
...
+            }
```
Comment (royjit) [inline docs/lib/auth/fragments/ios/user_attributes/10_fetch_attributes.md:17]:
```suggestion
if let email = attributes.first(where: { $0.key == .email }) {
     print("User email =  \(email).")
}
```

File: docs/lib/auth/fragments/android/user_attributes/10_fetch_attributes.md [hunk 3]
```diff
@@ -13,7 +25,19 @@ Amplify.Auth.fetchUserAttributes(
...
+        // To access a specific user attribute:
+        val attributesMap: MutableMap<AuthUserAttributeKey, String> = HashMap()
+        for (attribute in attributes) {
+            attributesMap[attribute.key] = attribute.value
+        }
+        val email = attributesMap[AuthUserAttributeKey.email()]
+        if (email != null) {
+            Log.i("AuthDemo", "User email = $email")
+        }
+    },
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/android/user_attributes/10_fetch_attributes.md:40]: More closely aligned to what @royjit suggested for Swift, how about:

```suggestion
    { attributes ->
        Log.i("AuthDemo", "All user attributes = $attributes")

        // To access a specific user attribute:
        val email = attributes
            .firstOrNull { it.key == AuthUserAttributeKey.email() }?.value
        if (email != null) {
            Log.i("AuthDemo", "User email = $email")
        }
    }, 
```

File: docs/lib/auth/fragments/android/user_attributes/10_fetch_attributes.md [hunk 4]
```diff
@@ -3,7 +3,19 @@
 
 ```java
```
Comment (jamesonwilliams) [inline docs/lib/auth/fragments/android/user_attributes/10_fetch_attributes.md:4]: Is there an RxJava example for this API, too? Could do like:
```java
Amplify.Auth.fetchUserAttributes()
    .filter(attribute -> AuthUserAttributesKey.email().equals(attribute.getKey()))
    .subscribe(email -> Log.i("AuthDemo", "User email = " + email.getValue()));
```

**Pull request closed without merging**